src/ElmFormat.hs

```diff
@@ -133,48 +155,42 @@ determineDestination output doValidate json =
 determineWhatToDo :: Source -> Destination -> Either ErrorMessage WhatToDo
 determineWhatToDo source destination =
     case ( source, destination ) of
-        ( Stdin, ValidateOnly ) -> Right $ ValidateStdin
+        ( Stdin version, ValidateOnly ) -> Right $ ValidateStdin version
         ( FromFiles first rest, ValidateOnly) -> Right $ ValidateFiles first rest
-        ( Stdin, UpdateInPlace ) -> Right StdinToStdout
-        ( Stdin, ToJson ) -> Right StdinToJson
-        ( Stdin, ToFile output ) -> Right $ StdinToFile output
+        ( Stdin version, UpdateInPlace ) -> Right $ StdinToStdout version
+        ( Stdin version, ToJson ) -> Right $ StdinToJson version
+        ( Stdin version, ToFile output ) -> Right $ StdinToFile version output
         ( FromFiles first [], ToFile output ) -> Right $ FormatToFile first output
         ( FromFiles first rest, UpdateInPlace ) -> Right $ FormatInPlace first rest
         ( FromFiles _ _, ToFile _ ) -> Left SingleOutputWithMultipleInputs
         ( FromFiles first [], ToJson ) -> Right $ FileToJson first
         ( FromFiles _ _, ToJson ) -> Left SingleOutputWithMultipleInputs
 
 
-determineWhatToDoFromConfig :: Flags.Config -> Either [InputFileMessage] [FilePath] -> Either ErrorMessage WhatToDo
-determineWhatToDoFromConfig config resolvedInputFiles =
+determineWhatToDoFromConfig :: Flags.Config -> ElmVersion -> Either [InputFileMessage] [ElmFile] -> Either ErrorMessage WhatToDo
+determineWhatToDoFromConfig config defaultElmVersion resolvedInputFiles =
     do
-        source <- determineSource (Flags._stdin config) resolvedInputFiles
+        checkUpgradeVersion (Flags._upgrade config) (Flags._elmVersion config)
```

Comment:
In 0.8.2, upgrade can actually use the auto-detected current directory elm version, which seems confusing to me (ie: `--upgrade ` is possible without `--elm-version`):

`MustSpecifyVersionWithUpgrade` says:
> "I can only upgrade code to specific Elm versions.  To make sure I'm doing what you expect, you must also specify --elm-version=" ++ show elmVersion ++ " when you use --upgrade."


This check makes sure that a flag is provided, whatever the result of "current directory elm version" (which might be removed) or "per file version detection" (which is overridden by the flag).

I'm not sure if this is the expected behavior.